packages/node/src/indexer/sandbox.service.ts

```diff
@@ -31,11 +34,16 @@ export class SandboxService<Api> {
       ? this.storeService.getStore()
       : hostStoreToStore((global as any).host); // Provided in worker.ts
 
+    const cache: Cache = isMainThread
+      ? this.cacheService.getCache()
+      : hostCacheToCache((global as any).host);
```

Comment:
Can we use the approach used by unfinalized blocks service instead please. We should inject the worker version in worker-fetch.module.ts

Ideally we would do the same with the store serverice